csrc/transformer/inference/csrc/pt_binding.cpp

```diff
@@ -352,15 +354,10 @@ void attention_unfused(T* prev_key_cont,
     float layer_scale = alibi.sizes().size() > 1 ? std::max(1, layer_id) : 1.0;
     float alpha = norm_factor * norm_factor / layer_scale;
     float gemm_beta = 0.0;
-    T* workspace;
-    if (seq_len == 1) {
-        workspace = (T*)output + bsz * seq_len * heads * k;
-    } else {
-        // If we are doing the prompt, switch to the tail workspace
-        T* scratch = (T*)Context::Instance().GetWorkSpace();
-        workspace = scratch + ((Context::Instance().get_workspace_size() / sizeof(T)) -
-                               bsz * heads * seq_len * soft_len);
-    }
+    // Always use the tail workspace
+    T* scratch = (T*)Context::Instance().GetWorkSpace();
+    T *workspace = scratch + ((Context::Instance().get_workspace_size() / sizeof(T)) -
+                            bsz * heads * seq_len * soft_len);
```

Comment:
we should divide the number of heads by mp_size so as not to request for more memory.